llvmlite/binding/lld.py

```diff
@@ -0,0 +1,85 @@
+from ctypes import c_bool, c_uint, POINTER, c_char_p, c_int, c_wchar_p
+import platform
+
+from llvmlite.binding import ffi
+
+ffi.lib.lld_main.restype = c_bool
+ffi.lib.lld_main.argtypes = [c_int, POINTER(c_char_p), POINTER(c_char_p)]
+def lld_main(lld_args) -> str:
+    """
+    Calls LLD - the LLVM linker.
+
+    The function calls the `lld::elf::link()` with the `lld_args` (list of
+    strings) as arguments.
+
+    Examples
+    ~~~~~~~~
+
+    Shows help:
+
+    >>> from llvmlite.binding import lld_main
+    >>> lld_main(["ld.lld", "--help"])
+    """
+    args = (c_char_p*len(lld_args))()
+    for i, arg in enumerate(lld_args):
+        args[i] = arg.encode()
+    with  ffi.OutputString() as outstr:
+        r = ffi.lib.lld_main(len(lld_args), args, outstr)
+        if r != False:
+            raise Exception("lld_main() failed, error code: %d" % r)
+        
+        return str(outstr)
+
+def lld_windows(output: str, objects: list[str], args: list[str] = []) -> str:
```

Comment:
its possible to create an `lld_runner` class or something like that.

It would probably look like:
```py
class lld_runner:
    __slots__ = ('command',)

    def __init__(self, command: str):
        self.command = command

    def __call__(self, *args): # too lazy to put all the args rn
        lld_main([self.command, *args]) # fill this out correctly

lld_window = lld_runner("lld-link")
lld_linux = lld_runner("ld.lld")
...
```

I'm not sure of another easy way to eliminate it. Do you have any ideas?

Reply:
I just realized another option could be nested functions

```python
def lld_function(command: str) -> Callable[[str, list[str], list[str]], str]:
    def wrapped(output: str, objects: list[str], args = []) -> str:
        return lld_main([command, '-o', output, *objects, *args])
    return wrapped

lld_windows = lld_function("lld-link")
lld_linux = lld_funtion("ld.lld")
...
```

which one do you think looks better or is more readable?